code/game/objects/structures/crates_lockers/closets/__closet.dm

```diff
@@ -50,7 +50,7 @@ var/global/list/closets = list()
 /obj/structure/closet/LateInitialize(mapload, ...)
 	var/list/will_contain = WillContain()
 	if(will_contain)
-		create_objects_in_loc(src, will_contain)
+		create_objects_in_loc(opened && loc || src, will_contain)
```

Comment:
Ideally this would just use onDropInto, but that needs an atom passed. Considered making `create_objects_in_loc` use dropInto but that'd be a bit of a headache as well.